common/shield_exceptions.cc

```diff
@@ -75,6 +75,25 @@ bool IsWhitelistedReferrer(const GURL& firstPartyOrigin,
     }
   }
 
+  static std::map<GURL, std::vector<URLPattern> > whitelist_patterns_map = {{
+      firstPartyOrigin, {
```

Comment:
maybe I'm missing something, but if we're adding this by firstPartyOrigin and then looking it up by firstPartyOrigin, isn't it the same as `whitelist_patterns` below?

Reply:
whitelist patterns is simply a list of subresources that are allowed anywhere.
This is scoped to a certain origin only.